crates/sdk/src/action.rs

```diff
@@ -109,6 +116,10 @@ impl<'a> Prove<'a> {
             core_opts: SP1CoreOpts::default(),
             recursion_opts: SP1CoreOpts::recursion(),
             timeout: None,
```

Comment:
note: the actions prove struct, context builder, and proof opts seem kinda clunky to me (at least for using things on the network) - for example, it's not clear why max_cycles is not simply a field

but refactoring all this is out of scope of this PR, no users really see this anyway